### PR TITLE
feat: Token create & delete from CLI

### PR DESCRIPTION
**Pull request by codekeyz**
Resolves: https://github.com/invertase/dart_globe/issues/134

### CODE REVIEW

**Comment by codekeyz** [comment]
## Creating Token

```shell
$ globe token create --name="Token" --expiry="2012-02-12" --project 8325e5a6-e419-45cb-99bd-3549880b1b26
```

or 

```shell
$ globe token create --name="Token" --expiry="2012-02-12"
Select projects to associate token with:
❯ ◯  pizza-store
  ◯  zomato
  ◯  blog-backend
```

### Terminal display when token created

```shell
$ globe token create --name="Token" --expiry="2012-02-12"
Select projects to associate token with: [zomato, pizza-app]
✓ Here's your token:
ID: 5c65febd-0185-4dca-9032-a4e3bdc8be4b
Token: foo-bar.token-goes.here (1.8s)
```

### Terminal display when invalid expiry format

```shell
$ globe token create --name="Token" --expiry="2012"
Invalid date format.
Date format should be 2012-02-27 or 2012-02-27 13:27:00
```

## Deleting Token

```shell
$ globe token delete --tokenId="5c65febd-0185-4dca-9032-a4e3bdc8be4b"
```

https://github.com/invertase/globe/assets/22515508/e8261847-3fcd-45d9-a30b-05470dd4e325

**Comment by codekeyz** [comment]
Blocked by: https://github.com/invertase/dart_globe/issues/159

File: packages/globe_cli/lib/src/commands/token/token_create_command.dart [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  String get name => 'create';
+
+  @override
+  FutureOr<int> run() async {
+    requireAuth();
+
+    final name = argResults?['name'] as String;
+    final projectIds = argResults?['project'] as List<String>?;
+    final dateString = argResults?['expiry'] as String;
+    final expiry = DateTime.tryParse(dateString);
```
Comment (lesnitsky) [inline packages/globe_cli/lib/src/commands/token/token_create_command.dart:44]: could we make all these optional and query for those if not provided via flag?

**Review by lesnitsky — CHANGES_REQUESTED** [review]

File: packages/globe_cli/lib/src/commands/token/token_delete_command.dart [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  String get description => 'Delete globe auth token.';
+
+  @override
+  String get name => 'delete';
+
+  @override
+  FutureOr<int> run() async {
+    requireAuth();
+
+    final tokenId = argResults?['tokenId'] as String;
```
Comment (lesnitsky) [inline packages/globe_cli/lib/src/commands/token/token_delete_command.dart:26]: same as https://github.com/invertase/globe/pull/39/files#r1494721331

**Comment by lesnitsky** [comment]
[`docs/cli/commands`](https://github.com/invertase/globe/tree/main/docs/cli/commands) should mention this new API as well.

**Comment by lesnitsky** [comment]
We'd also need a way to list tokens and see what projects a specific token has access to. Do you want to take care of it in this PR or separately?

**Comment by codekeyz** [comment]
> We'd also need a way to list tokens and see what projects a specific token has access to. Do you want to take care of it in this PR or separately?

Yeah, we'll have it in here.

**Review by lesnitsky — APPROVED** [review]

File: docs/cli/commands/token.mdx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+You can run the command interactively by running 
+
+```bash
+globe token create
+```
+
+or in-lined by providing necessary arguments 
+
+- `--name`- specify name to identity token.
```
Comment (exaby73) [inline docs/cli/commands/token.mdx:21]:
```suggestion
- `--name`- specify name to identity the token.
```

File: docs/cli/commands/token.mdx [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+- `--expiry` - specify lifespan of token.
```
Comment (exaby73) [inline docs/cli/commands/token.mdx:22]:
```suggestion
- `--expiry` - specify lifespan of the token.
```

File: docs/cli/commands/token.mdx [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+- `--project` - specify projects(s) to associate token with.
```
Comment (exaby73) [inline docs/cli/commands/token.mdx:23]:
```suggestion
- `--project` - specify projects(s) to associate the token with.
```

File: docs/cli/commands/token.mdx [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+```
+
+# Delete Token
+
+The `delete` command allows you to delete token by providing token ID.
+
+## Usage
+
+```bash
+globe token delete
```
Comment (exaby73) [inline docs/cli/commands/token.mdx:46]: The example here doesn't have a token ID

**Review by exaby73 — CHANGES_REQUESTED** [review]

File: docs/cli/commands/token.mdx (hunk 6 above)
Comment (codekeyz) [reply docs/cli/commands/token.mdx]: Resolved.

**Review by exaby73 — APPROVED** [review]
LGTM

**Pull request merged**